lib/nodejs/worker.ex

```
@@ -123,6 +123,15 @@ defmodule NodeJS.Worker do
     end
   end
 
+  def handle_info({_pid, data}, state) do
+    with :dev <- Mix.env(),
+         {_, {:eol, msg}} <- data do
+      IO.puts("NodeJS: #{msg}")
+    end
+
+    {:noreply, state}
+  end
```

There are a couple of things that concern me about this change as-is:
- I believe the call to `Mix.env()` will raise an exception in any application shipped as a release, since `Mix` isn't included / started in releases
- If the library needs to emit log messages, it should use `Logger` instead of `IO` so the authors of the application can have better control over where / whether those messages get logged

It seems like the big issue here is that your JS runtime is emitting extra messages on startup due to the debugger, which I think is just not foreseen or handled well by this library or generally by the stdin / stdout method of communication with the Port. Something like this _might_ work, but if the JS app were to emit text to stdout _during_ the execution of a function, you can see how that'd really mess with the `receive` loop above.

---

Thanks

I'll fix the mix.env() and the IO call. 

For the rest, I don't understand what's your concern. These messages are currently not processed by the library, and we get runtime errors because the handle_info callback is not implemented. I'm just suppressing these errors without affecting at all how the library currently works

---

I changed the IO call to use the logger, and guarded against Mix.env() crashing in release mode. 
I think it's safe this way - again, these messages are not meant to be processed by the library.

Let me know if it's fine like this or if you'd like some other changes

---

Hi @francois-codes,

First, we sincerely apologize for the long wait on this PR. Our team is short-handed and focused on client work, making it difficult to give our open source projects the attention they deserve.

We're merging your PR as it addresses an important issue blocking projects like live_vue. While your solution with Mix.env() rescue works, there are some concerns with this approach in releases - custom release configurations might handle module loading differently, future Elixir versions could change Mix behavior, and there's a small performance hit from triggering rescues on every call.

We'll follow up with a small refactor to use Application configuration instead:

```elixir
defp debug_mode? do
  Application.get_env(:nodejs, :debug_mode, false)
end
```

This provides a more reliable approach that's consistent with Elixir best practices.

Thanks for your contribution and understanding regarding our response time!

---

FYI https://github.com/revelrylabs/elixir-nodejs/pull/99

We will get a new release out once this gets in. Probably wont be until monday.

---

👍 @Dacello big thank you for your work and transparency! 👐